staging/src/k8s.io/client-go/tools/clientcmd/client_config.go

```diff
@@ -671,6 +671,13 @@ func BuildConfigFromFlags(masterUrl, kubeconfigPath string) (*restclient.Config,
 		}
 		klog.Warning("error creating inClusterConfig, falling back to default config: ", err)
 	}
+
+	if strings.Contains(kubeconfigPath, ":") {
+		configPaths := strings.Split(kubeconfigPath, ":")
```

Comment:
This isn't correct on Windows.

https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/#the-kubeconfig-environment-variable

Reply:
Can we re-use our existing implementation of splitting this?

Reply:
The variable name is also misleading now, and the function documentation should be updated?

Reply:
Yes, I missed it earlier. How about looking for the KUBECONFIG environment variable in code itself without waiting for the user to pass it? This way there is no need of changing the variable name and user will simply has to pass `BuildConfigFromFlags("", "")` .

Reply:
I can't personally comment on if the OWNERS of the client would agree to this approach. I do know that we should be careful to keep the behavior in sync closely.